src/app/services/log.service.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { TestBed } from '@angular/core/testing';
+
+import { LogService } from './log.service';
+
+describe('LogService', () => {
+  let service: LogService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    service = TestBed.inject(LogService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should output the given message as an error to the console', () => {
+    const error = jest.spyOn(console, 'error').mockImplementation();
+    const errorMessage = "This is an error message";
+    service.error(errorMessage);
+    expect(error).toHaveBeenCalled();
+    expect(error).toBeCalledWith('%c' + errorMessage, 'color: #DC143C');
+    error.mockReset();
+  });
+
+  it('should output the given message as a warning to the console', () => {
+    const warn = jest.spyOn(console, 'warn').mockImplementation();
+    const warningMessage = "This is an warning message";
```

Comment:
It's a minor pattern, but I'm a huge fan of it. I really like to see people putting strings/values into variables and then using those variables in the assertions. It makes maintaining much easier! 👏 👏 👏